libs/shared/util-auth/src/Operation.ts

```diff
@@ -1,5 +1,13 @@
 import { ProgramError } from "@banklessdao/util-data";
 import * as TE from "fp-ts/lib/TaskEither";
 
-export type Operation<I, O> = (input: I) => TE.TaskEither<ProgramError, O>;
-
+/**
+ * Represents an asynchronous operation that takes an arbitrary `input` and returns
+ * a wrapped result, that's either a success or a failure. Failure is represented by a
+ * [[ProgramError]] object, success is represented by an arbitrary object
+ * of type `O`.
+ */
+export type Operation<I, O> = {
```

Comment:
It is **much easier** to look up `Operation`s using their name and this also makes it possible to store the authorization information in a database without having to reference actual `Operation`s (we only need to refer to them by name now).